include/quic/parser.hpp

```diff
@@ -0,0 +1,384 @@
+#include <oxenc/bt.h>
+
+#include "stream.hpp"
+#include "utils.hpp"
+
+namespace oxen::quic
+{
+    inline auto bp_cat = oxen::log::Cat("bparser");
+
+    using time_point = std::chrono::steady_clock::time_point;
+
+    // timeout is used for sent requests awaiting responses
+    const std::chrono::seconds TIMEOUT{10};
+
+    // request sizes
+    inline constexpr long long MAX_REQ_LEN{10000000};
+
+    // Application error
+    inline constexpr uint64_t BPARSER_EXCEPTION = (1ULL << 60) + 69;
+
+    inline std::basic_string<std::byte> operator""_bs(const char* __str, size_t __len) noexcept
+    {
+        return std::basic_string<std::byte>(reinterpret_cast<const std::byte*>(__str), __len);
+    }
+
+    inline std::string_view to_sv(bstring_view x)
+    {
+        return {reinterpret_cast<const char*>(x.data()), x.size()};
+    }
+
+    struct message
+    {
+        std::string data;
+        std::string_view req_type;
+        std::string_view req_id;
+        std::string_view endpoint;
+        std::string_view req_body;
+
+        message(std::string req) : data{std::move(req)}
+        {
+            oxenc::bt_list_consumer btlc(data);
+
+            try
+            {
+                req_type = btlc.consume_string_view();
+                req_id = btlc.consume_string_view();
+
+                if (req_type == "Q" || req_type == "C")
+                    endpoint = btlc.consume_string_view();
+
+                req_body = btlc.consume_string_view();
+            }
+            catch (...)
+            {
+                log::critical(bp_cat, "Invalid request body!");
+            }
+        }
+
+        std::string rid() { return std::move(std::string{req_id}); }
```

Comment:
No need for the `move` here just `return std::string{req_id};` will do the trick.